src/posix/platform/system.cpp

```diff
@@ -175,11 +163,45 @@ void platformInit(otPlatformConfig *aPlatformConfig)
     ot::Posix::Udp::Get().Init(aPlatformConfig->mInterfaceName);
 #endif
 #endif
-
 exit:
     return;
 }
 
+void _platformInitOffloadMode(otPlatformConfig *aPlatformConfig)
+{
+    // Do nothing now.
+    OT_UNUSED_VARIABLE(aPlatformConfig);
+}
+
+void platformInit(otPlatformConfig *aPlatformConfig)
+{
+#if OPENTHREAD_POSIX_CONFIG_BACKTRACE_ENABLE
+    platformBacktraceInit();
+#endif
+
+    platformAlarmInit(aPlatformConfig->mSpeedUpFactor, aPlatformConfig->mRealTimeSignal);
+
+    sCoprocessorType = platformSpinelManagerInit(get802154RadioUrl(aPlatformConfig));
+
+    switch (sCoprocessorType)
+    {
+    case OT_COPROCESSOR_RCP:
+        _platformInitCoreMode(aPlatformConfig);
+        break;
+
+    case OT_COPROCESSOR_NCP:
+        _platformInitOffloadMode(aPlatformConfig);
```

Comment:
It seems that using the function name `platformInitRcpMode` and `platformInitNcpMode` is more consistent with these cases.